location/lbs/activity/audioset/dataprocessing/audio_processing.py

```diff
@@ -0,0 +1,466 @@
+end_time_seconds, and labels
+The downloaded audios would be clipped into the labelled 10 seconds segments
+The audio segments are then processed using Librosa to extract features
+
+Usage
+Standalone script:
+python audio_processing.py --src_dir PATH_TO_INPUT_FILES --des_dir DEST_DIR PATH_TO_OUTPUT_FILES_--audioset_csv CSV FILE --labels LABELS --redo BOOLEAN TO REDOWNLOAD THE YOUTUBE FILES
+example:
+python audio_processing.py --src_dir location/lbs/activity/audioset/dataprocessing/example_src_dir --des_dir location/lbs/activity/audioset/dataprocessing/example_dest_dir_--audioset_csv balanced_train_segments --labels "Gunshot, gunfire" --False
+
+Build with bazel:
+bazel build location/lbs/activity/audioset/dataprocessing:audio_processing
+bazel-bin/location/lbs/activity/audioset/dataprocessing/audio_processing --src_dir location/lbs/activity/audioset/dataprocessing/example_src_dir --des_dir location/lbs/activity/audioset/dataprocessing/example_dest_dir_--audioset_csv balanced_train_segments --labels "Gunshot, gunfire" --False
+
+
+--src_dir is the path to the directory containing the input files
+(audioset csv file ontology.json)
+--dest_dir is the path to the directory containing the output files
+(output csv file)
+--labels included in the AudioSet dataset can be used (i.e "Gunshot, gunfire")
+
+Performance: to confirm the presence of, extract features from, and output to a
+    csv file, takes about ___minutes for around 20,000 samples.
+"""
+
+from __future__ import unicode_literals
+import youtube_dl
+import csv
+import json
+from pydub import AudioSegment
+import os
+import shutil
+from os import listdir
+from os.path import isfile, join
+import librosa
+from absl import app
+from absl import flags
+from absl import logging
+import datetime
+
+FLAGS = flags.FLAGS
+
+
+def setup_abseil():
+    """Sets up command line arguments using Abseil.
+
+    Defines --audioset_csv, --labels, --redo, --src_dir, and --dest_dir as flags
+    """
+    flags.DEFINE_string('audioset_csv', 'balanced_train_segments',
+                        'The csv file of a dataset provided by the audioset')
+    flags.DEFINE_multi_string('labels', 'Gunshot, gunfire',
+                              'The set of labels identified by the audioset to '
+                              'treat as positive')
+    flags.DEFINE_bool('redo', False,
+                      'Whether to redownload the YouTube videos in the included'
+                      ' csv file')
+    flags.DEFINE_string('src_dir',
+                        'location/lbs/activity/audioset/dataprocessing/example_src_dir',
+                        'The location of the audioset csv file(s) and the '
+                        'ontology.json file')
+    flags.DEFINE_string('dest_dir',
+                        'location/lbs/activity/audioset/dataprocessing/example_dest_dir',
+                        'The location of the csv file and downloaded YouTube '
+                        'videos')
```

Comment:
typically we just follow the pattern at
https://abseil.io/docs/python/guides/flags

and define the flags in the outermost space.

Reply:
Ok. Thanks! I'll change that.